investors/serializers.py

```diff
@@ -38,3 +41,37 @@ def validate_company_name(self, value):
     def create(self, validated_data):
         validated_data['user'] = self.context['request'].user
         return super().create(validated_data)
+
+
+
+class SavedStartupSerializer(serializers.ModelSerializer):
+    startup = StartupSerializer(read_only=True)
+    startup_id = serializers.PrimaryKeyRelatedField(
+        queryset=Startup.objects.all(), source='startup', write_only=True
+    )
+
+    class Meta:
+        model = SavedStartup
+        fields = ['id', 'investor_id', 'startup', 'startup_id', 'status', 'notes', 'saved_at', 'created_at', 'updated_at']
+        read_only_fields = ['id', 'saved_at', 'created_at', 'updated_at']
+
+    def validate(self, attrs):
+        request = self.context.get('request')
+        investor = getattr(request.user, 'investor', None) if request else None
+        startup = attrs.get('startup')
+
+        if not investor:
+            raise serializers.ValidationError({'detail': 'Only investors can save startups.'})
+
+        if hasattr(request.user, 'startup') and request.user.startup_id == getattr(startup, 'id', None):
+            raise serializers.ValidationError({'startup_id': 'You cannot save your own startup.'})
+
+        if SavedStartup.objects.filter(investor=investor, startup=startup).exists():
```

Comment:
This check can cause a race condition in concurrent requests. Consider using a database-level unique constraint (already present in your model) and handling IntegrityError on save for robustness.